#! /bin/sh /usr/share/dpatch/dpatch-run
## call_ld_with_proper_flags.dpatch by Julien Cristau <julien.cristau@ens-lyon.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Filter linkflags passed to ld by ocamlopt -pack and -output-obj
## DP: to remove the "-Wl," parts, which are only used when ocamlopt calls
## DP: gcc.

@DPATCH@
diff -urNad ocaml-3.09.1+3.09.2rc1~/Makefile ocaml-3.09.1+3.09.2rc1/Makefile
--- ocaml-3.09.1+3.09.2rc1~/Makefile	2006-03-30 17:40:58.000000000 +0200
+++ ocaml-3.09.1+3.09.2rc1/Makefile	2006-03-30 17:41:00.000000000 +0200
@@ -313,6 +313,7 @@
 partialclean::
 	rm -f ocaml toplevel/toplevellib.cma
 
+Wl = -Wl,
 # The configuration file
 
 utils/config.ml: utils/config.mlp config/Makefile
@@ -324,8 +325,8 @@
             -e 's|%%BYTELINK%%|$(BYTECC) $(BYTECCLINKOPTS)|' \
             -e 's|%%NATIVECC%%|$(NATIVECC) $(NATIVECCCOMPOPTS)|' \
             -e 's|%%NATIVELINK%%|$(NATIVECC) $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PARTIALLD%%|ld -r $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PACKLD%%|ld -r $(NATIVECCLINKOPTS)|' \
+            -e 's|%%PARTIALLD%%|ld -r $(subst $(Wl),,$(NATIVECCLINKOPTS))|' \
+            -e 's|%%PACKLD%%|ld -r $(subst $(Wl),,$(NATIVECCLINKOPTS))|' \
             -e 's|%%BYTECCLIBS%%|$(BYTECCLIBS)|' \
             -e 's|%%NATIVECCLIBS%%|$(NATIVECCLIBS)|' \
             -e 's|%%RANLIBCMD%%|$(RANLIBCMD)|' \
